### PR TITLE
Emit a success factor from maintainers

### PR DESCRIPTION
**Pull request by bratseth**

### CODE REVIEW

**Comment by bratseth** [comment]
@mpolden please review

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/ScalingSuggestionsMaintainer.java [hunk 1]
```diff
@@ -36,13 +36,16 @@ public ScalingSuggestionsMaintainer(NodeRepository nodeRepository,
...
 
+        int attempts = 0;
         int successes = 0;
-        for (var application : activeNodesByApplication().entrySet())
+        for (var application : activeNodesByApplication().entrySet()) {
+            attempts++;
             successes += suggest(application.getKey(), application.getValue());
-        return successes > 0;
+        }
+        return attempts == 0 ? 1.0 : ((double)successes / attempts);
```
Comment (mpolden) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/ScalingSuggestionsMaintainer.java:48]: Use `asSuccessFactor` here too?

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/RetiredExpirer.java [hunk 2]
```diff
@@ -57,17 +60,19 @@ protected boolean maintain() {
...
                 boolean success = deployment.activate().isPresent();
-                if ( ! success) return success;
+                if ( ! success) continue;
                 String nodeList = nodesToRemove.stream().map(Node::hostname).collect(Collectors.joining(", "));
                 log.info("Redeployed " + application + " to deactivate retired nodes: " +  nodeList);
+                successes++;
             }
         }
-        return true;
+        return attempts == 0 ? 1.0 : ((double)successes / attempts);
```
Comment (mpolden) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/RetiredExpirer.java:75]: Use `asSuccessFactor` here too?

**Review by mpolden — APPROVED** [review]
Great!

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/RetiredExpirer.java (hunk 2 above)
Comment (bratseth) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/RetiredExpirer.java:75]: Here is was easier to count successes than failures.

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/ScalingSuggestionsMaintainer.java (hunk 1 above)
Comment (bratseth) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/ScalingSuggestionsMaintainer.java:48]: Here is was easier to count successes than failures.

**Pull request merged**